tests/network/vmi_networking.go

```diff
@@ -204,30 +203,26 @@ var _ = SIGDescribe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:c
 
 				ip := inboundVMI.Status.Interfaces[0].IP
 
-				tcpJob := job.NewHelloWorldJobTCP(ip, strconv.Itoa(testPort))
-				tcpJob.Spec.Template.Spec.Affinity = &k8sv1.Affinity{
-					NodeAffinity: &k8sv1.NodeAffinity{
-						RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
-							NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
-								{
-									MatchExpressions: []k8sv1.NodeSelectorRequirement{
-										{Key: k8sv1.LabelHostname, Operator: op, Values: []string{inboundVMI.Status.NodeName}},
-									},
-								},
-							},
-						},
-					},
-				}
-				tcpJob.Spec.Template.Spec.HostNetwork = hostNetwork
+				By("start connectivity job on the same node as the VM")
+				localNodeTCPJob := job.NewHelloWorldJobTCP(ip, strconv.Itoa(testPort))
+				localNodeTCPJob.Spec.Template.Spec.HostNetwork = hostNetwork
+				localNodeTCPJob.Spec.Template.Spec.Affinity = &k8sv1.Affinity{NodeAffinity: newNodeAffinity(inboundVMI.Status.NodeName)}
+				localNodeTCPJob, err = virtClient.BatchV1().Jobs(inboundVMI.ObjectMeta.Namespace).Create(context.Background(), localNodeTCPJob, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
 
-				tcpJob, err = virtClient.BatchV1().Jobs(inboundVMI.ObjectMeta.Namespace).Create(context.Background(), tcpJob, metav1.CreateOptions{})
+				By("start connectivity job on different node")
+				remoteNodeTCPJob := job.NewHelloWorldJobTCP(ip, strconv.Itoa(testPort))
+				remoteNodeTCPJob.Spec.Template.Spec.HostNetwork = hostNetwork
+				remoteNodeTCPJob.Spec.Template.Spec.Affinity = &k8sv1.Affinity{PodAntiAffinity: newPodAntiAffinity(testLabel)}
```

Comment:
What is the reason you had to change this to a label affinity from a node (anti) affinity?
I prefer you change one thing at a time, it makes it easier to review and you can reason for it well in the commit/PR.

Reply:
Initially it made more sense to me using pod-anti-affinity and not care for the node.
Looking at this again we can still use node affinity rules, I can revert it to how it was.

Reply:
The change from the original and the asymmetric usage does not make sense to me without a proper reasoning.

> not care for the node

I do not know what this means. We care about the nodes because we need these to run on different nodes.